elastic/kibana
# [Uptime] Monitor Details - fallback to saved objects when monitor details are unavailable from last ping

## Summary

For the Monitor Details page, this PR falls back to checking monitor saved objects when a recent document is not available within the selected time range.

Thanks to UI Monitor Management, we now have basic information about a monitor, even when the monitor pings are outside of the selected time range. To take advantage this, we check the saved objects when a recent ping is not available, and use any available data from the monitor saved object to build out the page.

In the below screenshot, the **monitor name** at the top (which I uninspiredly called My Special Monitor), the **monitor type**, and the **monitor URL** is pulled from the saved object. 
![image](https://user-images.githubusercontent.com/11356435/153601849-925a1f60-837a-4669-b4e3-ec03ccd2e14b.png)

### Testing
1. Create a monitor. **Note**: you do not need to be connected to the synthetics service for this, as this PR is specifically for handling the situation in which there **is not** data in the given time range. If you are connected to the service locally, it may be good to give your monitor a high time range, like 20 minutes.
2. After navigating to the monitor table, click the monitor name. Note: If you're connected to the synthetics service, it's best to wait until after you're sure any data would have called out of the 15 minute default timerange, or set the timerange to something super small
3. Verify that the monitor name, monitor type, and monitor URL appear. **Note**: For browser monitors, monitor URLS are hydrated into the saved objects async. So if you want to see the URL for a browser monitor, you'll need to actually be connected to the service so that the saved object can be hydrated.
